### DISCUSSION TITLE
Does this only applicable for certain devices?

### DISCUSSION THREAD

**Discussion by sameerasw** [reiryuki/Proximity-Sensor-Disabler-Magisk-Module#1] [Q&A] [answered]
I tested on my Pixel 4a 5G (because of the faulty sensor) but even if the prop updated to [false] the sensor is still active and behaves like before... Even the HW/sensor monitoring apps reports the same.

Thank you <3
![image](https://github.com/reiryuki/Proximity-Sensor-Disabler-Magisk-Module/assets/68902530/70b72705-eff9-441a-85eb-7fbff558d80c)
![image](https://github.com/reiryuki/Proximity-Sensor-Disabler-Magisk-Module/assets/68902530/3dddf472-4011-4e77-bdbc-83fb0a7cf21b)

**Comment by reiryuki** [comment]
send z folder https://t.me/androidryukimodsdiscussions/2618

and output of:

su

`strings /vendor/lib*/*sensor* | grep proximity`

**Reply by sameerasw** [reply]
[z.zip](https://github.com/reiryuki/Proximity-Sensor-Disabler-Magisk-Module/files/13703190/z.zip)
[output.txt](https://github.com/reiryuki/Proximity-Sensor-Disabler-Magisk-Module/files/13703193/output.txt)

output:

```
bramble:/ $ su
bramble:/ # strings /vendor/lib*/*sensor* | grep proximity
_ZN9proximityC2E10sensor_uid18sensor_wakeup_type
_ZTV9proximity
_ZN9proximity23handle_sns_client_eventERK37sns_client_event_msg_sns_client_event
_ZN19sns_proximity_eventC2EPN6google8protobuf5ArenaEb
_ZN19sns_proximity_eventD1Ev
_ZN9proximityC1E10sensor_uid18sensor_wakeup_type
_ZN16proximity_fusionC2E10sensor_uid18sensor_wakeup_type
_ZTV16proximity_fusion
_ZN16proximity_fusion23handle_sns_client_eventERK37sns_client_event_msg_sns_client_event
_ZN16proximity_fusion22update_fused_hal_eventEyNS_22FusedEventUpdateReasonE
_ZN16proximity_fusion24debouncing_timer_settimeEi
_ZN16proximity_fusion28handle_touch_prox_palm_eventERKN7android8hardware7sensors4V1_05EventE
_ZN16proximity_fusion20on_call_state_changeEN16CallStateMonitor9CallStateE
_ZN16proximity_fusion30on_sensor_or_call_state_changeEv
_ZN16proximity_fusion20get_hal_event_statusERKN7android8hardware7sensors4V1_05EventE
_ZN16proximity_fusion23debouncing_timer_createEv
_ZN16proximity_fusion27acquire_debouncing_wakelockEb
_ZN16proximity_fusion26on_debouncing_timer_expireEv
_ZN16proximity_fusionC1E10sensor_uid18sensor_wakeup_type
android.sensor.proximity
proximity
vendor.sensor.proximity_fusion.enabled
proximity
proximity
_ZN9proximityC2E10sensor_uid18sensor_wakeup_type
_ZTV9proximity
_ZN9proximity23handle_sns_client_eventERK37sns_client_event_msg_sns_client_event
_ZN19sns_proximity_eventC2EPN6google8protobuf5ArenaEb
_ZN19sns_proximity_eventD1Ev
_ZN9proximityC1E10sensor_uid18sensor_wakeup_type
_ZN16proximity_fusionC2E10sensor_uid18sensor_wakeup_type
_ZTV16proximity_fusion
_ZN16proximity_fusion23handle_sns_client_eventERK37sns_client_event_msg_sns_client_event
_ZN16proximity_fusion22update_fused_hal_eventEmNS_22FusedEventUpdateReasonE
_ZN16proximity_fusion24debouncing_timer_settimeEi
_ZN16proximity_fusion28handle_touch_prox_palm_eventERKN7android8hardware7sensors4V1_05EventE
_ZN16proximity_fusion20on_call_state_changeEN16CallStateMonitor9CallStateE
_ZN16proximity_fusion30on_sensor_or_call_state_changeEv
_ZN16proximity_fusion20get_hal_event_statusERKN7android8hardware7sensors4V1_05EventE
_ZN16proximity_fusion23debouncing_timer_createEv
_ZN16proximity_fusion27acquire_debouncing_wakelockEb
_ZN16proximity_fusion26on_debouncing_timer_expireEv
_ZN16proximity_fusionC1E10sensor_uid18sensor_wakeup_type
android.sensor.proximity
proximity
vendor.sensor.proximity_fusion.enabled
bramble:/ # exit
bramble:/ $ exit
```

**Comment by reiryuki** [comment]
add

vendor.sensor.proximity_fusion.enabled=false

to /data/adb/modules/ProximitySensorDisabler/system.prop

and

/data/adb/modules_update/ProximitySensorDisabler/system.prop

**Reply by sameerasw** [reply]
I can edit the magisk module zip file content with these changes and re-flash right?

Maybe not.... I'll try another way


Thanks a lot <3 I will try

**Comment by reiryuki** [comment]
yep

**Reply by sameerasw** [reply]
I couldn't find 

> /data/adb/modules_update/ProximitySensorDisabler/system.prop

Since it didn't exist....

**Reply by sameerasw** [reply]
[proximity.zip](https://github.com/reiryuki/Proximity-Sensor-Disabler-Magisk-Module/files/13703964/proximity.zip)

this is the module I edited and try flashing but no difference

**Reply by reiryuki** [reply]
su

`strings /vendor/bin/hw/*sensor* | grep proximity`

**Reply by sameerasw** [reply]
No output
![image](https://github.com/reiryuki/Proximity-Sensor-Disabler-Magisk-Module/assets/68902530/cc41e01a-f262-4692-9dff-156d5e7e4cea)

**Reply by reiryuki** [reply, marked as answer]
try this : https://t.me/androidryukimodsdiscussions/214525

**Reply by sameerasw** [reply]
I'll try next time... I flashed the old script a couple of times, and it somehow magically fixed the sensor... Now everything works fine...

Thanks <3

**Reply by reiryuki** [reply]
which old version? and what do you mean with "fixed the sensor"? please describe it completely

**Reply by sameerasw** [reply]
I flashed the zip file with the changes you mentioned [here ](https://github.com/reiryuki/Proximity-Sensor-Disabler-Magisk-Module/discussions/1#discussioncomment-7885451) ....

My proximity sensor was locked at 0.0cm and AOD and such apps were not usable... Also in a call after the screen turn off, it never turned back on without the power key.... But when I did use the power key it turns off when I cover the sensor again. So the sensor was reporting only covering and not revealing....

The issue was fixed after me trying to disable the sensor by flashing the magisk module from [here](https://www.pling.com/p/2100679/) .... But not even a software reset fixed before.

**Reply by reiryuki** [reply]
ok then I will add
vendor.sensor.proximity_fusion.enabled=false
to the next version

**Reply by sameerasw** [reply]
Thanks!

**Reply by reiryuki** [reply]
please send output:

su

`strings /vendor/lib*/hw/*sensor* | grep proximity`

**Reply by sameerasw** [reply]
It outputs nothing but I just switched to a rom  too.... So the results might be different than the stock os I had before... 

But I remember digging into the tree and never found a directory named sensor in it... It was supposed to be there but instead I found some files in INPUT directory....

**Reply by reiryuki** [reply]
ok then

**Reply by reiryuki** [reply]
If you tried the beta next time, then tell me is it disable your proximity or not. I still need the result.

**Reply by sameerasw** [reply]
Ok sure... curretly on a KSU rom and it's not really good at modifying these stuff... I couldn't force voLTE and such with it unlike magisk... I'll try...

:)
